### DISCUSSION TITLE
I am not sure why this function is not being found upon request

### DISCUSSION THREAD

**Discussion by eslksw** [PostgREST/postgrest#2417] [Q&A] [answered]
So, I was trying to make a kind of complex PL/pgSQL function accessible via postgREST but kept failing and got always the error:

```json
{
    "hint": null,
    "details": null,
    "code": "42P01",
    "message": "relation \"api.globalquery\" does not exist"
}
```

However, my function does of course exists, as proven by curling /rpc/

```json
"/rpc/globalquery": {
            "post": {
                "tags": [
                    "(rpc) globalquery"
                ],
                "produces": [
                    "application/json",
                    "application/vnd.pgrst.object+json"
                ],
                "parameters": [
                    {
                        "required": true,
                        "schema": {
                            "required": [
                                "_tables",
                                "_filter"
                            ],
                            "type": "object",
                            "properties": {
                                "_filter": {
                                    "format": "json"
                                },
                                "_tables": {
                                    "format": "json"
                                }
                            }
                        },
                        "in": "body",
                        "name": "args"
                    },
                    {
                        "$ref": "#/parameters/preferParams"
                    }
                ],
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
```

I then tried with an example from the postgREST documentation. So I created the function

```sql
CREATE FUNCTION api.pplus(x integer, y integer)
RETURNS integer AS $$
 SELECT x + y;
$$ LANGUAGE SQL IMMUTABLE;
```

made it available to remote_user and then curled with

```shell
curl "https://localhost:3000/rpc/pplus" -X POST -H "Content-Type: application/json" -d '{ "x": 1, "y": 2 }'
```

and I keep getting the same error, although pplus also appears in the /rpc/ curl.

Anyone knows what I am doing wrong? I would be greatly appreciated.

**Comment by steve-chavez** [comment]
> So, I was trying to make a kind of complex PL/pgSQL function accessible via postgREST but kept failing and got always the error:

How did you call the `globalquery` function? (curl example)

**Reply by eslksw** [reply]
I did

```shell
curl "https://localhost:3000/rpc/globalquery" -X POST -H "Content-Type: application/json" -d '{ "_tables":  {"tablenames" : ["foo"]}, "_filter": {"conditions": {"bar_key" : "bar_value"}} }'
```

**Reply by steve-chavez** [reply]
Did you [reload the schema cache](https://postgrest.org/en/latest/schema_cache.html#schema-cache-reloading)?

**Reply by eslksw** [reply]
Yes, I used `killall -SIGUSR1 postgrest`, and I also tried `systemctl restart` on the custom postgREST service.

**Comment by laurenceisla** [comment, marked as answer]
The error you get is as if it's selecting from a non existent table from the database, like: `curl "http://localhost:3000/globalquery"`. When the function is not found the error is different. I noticed that you're using `https` which is not supported out of the box; perhaps a web server like Nginx is changing something there?.

Also, which version of PostgREST and PostgreSQL are you running?

**Reply by eslksw** [reply]
Yes, that might be it. HTTPS is required by the project head, so falling back to HTTP is not really an option. I will still try in order to see whether that's the problem, thanks for the hint!

**Reply by wolfgangwalther** [reply]
> HTTPS is required by the project head, so falling back to HTTP is not really an option.

Nobody suggested that. The `https` in your code snippet is an indication that you're not showing us the full picture here and there are some infos missing. That's what @laurenceisla was refering to.
